src/Libraries/RevitNodes/Elements/FloorType.cs

```diff
@@ -135,6 +112,53 @@ internal static FloorType FromExisting(Autodesk.Revit.DB.FloorType floorType, bo
 
         #endregion
 
+        #region Public methods
+
+        /// <summary>
+        /// Gets the structural material of the FloorType.
+        /// </summary>
+        /// <returns>Returns the material that defines the element's structural analysis properties.</returns>
+        public Element GetStructuralMaterial()
+        {
+            ElementId materialId = InternalFloorType.StructuralMaterialId;
+            if (materialId == null || materialId.IntegerValue < 0)
+                throw new InvalidOperationException(Properties.Resources.NoStructuralMaterialAssigned);
+            return Document.GetElement(materialId).ToDSType(true);
+        }
+
+        /// <summary>
+        /// The calculated and settable thermal properties of the FloorType
+        /// </summary>
+        /// <returns name = "Absorptance">Value of absorptance.</returns>
+        /// <returns name = "HeatTransferCoefficient">The heat transfer coefficient value (U-Value).</returns>
+        /// <returns name = "Roughness">Value of roughness.</returns>
+        /// <returns name = "ThermalMass">The calculated thermal mass value.</returns>
+        /// <returns name = "ThermalResistance">The calculated thermal resistance value (R-Value).</returns>
+        [MultiReturn(new[] { absorptanceOutputPort, heatTransferCoefficientOutputPort, roughnessOutputPort, thermalMassOutputPort, thermalResistanceOutputPort })]
+        public Dictionary<string, object> GetThermalProperties()
+        {
+            ThermalProperties thermalProperties = this.InternalFloorType.ThermalProperties;
+            if (thermalProperties == null)
+                throw new InvalidOperationException(nameof(GetThermalProperties));
+
+            double absorptance = thermalProperties.Absorptance;
+            double heatTransferCoefficient = thermalProperties.HeatTransferCoefficient;
+            double roughness = thermalProperties.Roughness;
+            double thermalMass = thermalProperties.ThermalMass;
+            double thermalResistance = thermalProperties.ThermalResistance;
```

Comment:
we should use these properties like `thermalProperties.Absorptance` directly and not allocate another `double` value just to return it.